src/main/java/com/iexec/sms/admin/AdminService.java

```diff
@@ -35,8 +45,62 @@ public AdminService(@Value("${spring.datasource.url}") String datasourceUrl,
         this.datasourcePassword = datasourcePassword;
     }
 
-    public String createDatabaseBackupFile() {
-        return "createDatabaseBackupFile is not implemented";
+    /**
+     * Creates a backup of the H2 database and saves it to the specified location.
+     *
+     * @param storageLocation The location where the backup file will be saved, must be an existing directory.
+     * @param backupFileName  The name of the backup file.
+     * @return {@code true} if the backup was successful; {@code false} if any error occurs.
+     */
+    boolean createDatabaseBackupFile(String storageLocation, String backupFileName) {
+        // Check for missing or empty storageLocation parameter
+        if (StringUtils.isBlank(storageLocation)) {
+            log.error("storageLocation must not be empty.");
+            return false;
+        }
+        // Check for missing or empty backupFileName parameter
+        if (StringUtils.isBlank(backupFileName)) {
+            log.error("backupFileName must not be empty.");
+            return false;
+        }
+        // Ensure that storageLocation ends with a slash
+        if (!storageLocation.endsWith("/")) {
+            storageLocation += "/";
```

Comment:
There is a `File.separator` constant value which is nice to know for such usages

Reply:
done in local, it's clearly better